deploy/buildtools/buildafi.py

```diff
@@ -164,17 +250,21 @@ def aws_build(global_build_config, bypass=False):
         agfi_afi_ids = local("""aws ec2 create-fpga-image --input-storage-location Bucket={},Key={} --logs-storage-location Bucket={},Key={} --name "{}" --description "{}" """.format(s3bucket, "dcp/" + s3_tarfile, s3bucket, "logs/", afiname, description), capture=True)
         rootLogger.debug(agfi_afi_ids)
         rootLogger.debug(agfi_afi_ids.stderr)
-        rootLogger.debug("create-fpge-image result: " + str(agfi_afi_ids))
+        rootLogger.debug("create-fpga-image result: " + str(agfi_afi_ids))
         ids_as_dict = json.loads(agfi_afi_ids)
         agfi = ids_as_dict["FpgaImageGlobalId"]
         afi = ids_as_dict["FpgaImageId"]
         rootLogger.info("Resulting AGFI: " + str(agfi))
         rootLogger.info("Resulting AFI: " + str(afi))
 
+    if global_build_config.distributed_elaboration:
+        # if we generated verilog (and driver headers) remotely, we now need to build the driver locally
+        rootLogger.info("Building FireSim-f1 driver")
```

Comment:
I'm assuming the timestamps workout such that only the C++ is being compiled here? 
We could also just build the driver in all cases and drop the if.

Reply:
Yes, the timestamps should be preserved if I remember correctly from reading the Fabric rsync'ing code but I'll double check.  I thought it was useful to have the comment there.  You're correct that it won't hurt anything to remove the if.  I can do that and adjust the comment.

Reply:
Mmm, that makes sense. I'll leave it to your discretion.

Reply:
Fabric 1.14 `rsync_project` uses `-pthrvz` as the default options. https://github.com/fabric/fabric/blob/1.14/fabric/contrib/project.py#L27

`-t` preserves the modification time and I don't see us passing in different default_opts to any of the calls or extra_opts that would change that.

I suppose there's always the chance that AWS system times may be inconsistent.  I've run into trouble with rsync'ing files correctly between hosts when the ntpd was jacked in the datacenter and having really weird things happen.  

I checked my firesim manager and noticed that it's setup using chrony pointing at the usual ntp.org servers.  It might be better to point the manager and build servers at the local Amazon Time Service by adding it's server info (`server 169.254.169.123 prefer iburst` as per https://aws.amazon.com/blogs/aws/keeping-time-with-amazon-time-sync-service/) to /etc/crony.conf and restarting chrony as a part of the startup script.  Hard to say whether it is worth it without starting to do some distributed builds and seeing if there are issues.  https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-time.html also has some useful info about Amazon Time Sync Service

Before adding the Amazon server, I saw:
```
bash-4.2$ chronyc sources -v
210 Number of sources = 4

  .-- Source mode  '^' = server, '=' = peer, '#' = local clock.
 / .- Source state '*' = current synced, '+' = combined , '-' = not combined,
| /   '?' = unreachable, 'x' = time may be in error, '~' = time too variable.
||                                                 .- xxxx [ yyyy ] +/- zzzz
||      Reachability register (octal) -.           |  xxxx = adjusted offset,
||      Log2(Polling interval) --.      |          |  yyyy = measured offset,
||                                \     |          |  zzzz = estimated error.
||                                 |    |           \
MS Name/IP address         Stratum Poll Reach LastRx Last sample               
===============================================================================
^* chl.la                        2   9   377   483   +478us[  +80us] +/-   12ms
^- 108.61.73.244                 2  10   377   673  -2898us[-3218us] +/-   54ms
^- 44.190.6.254                  3  10   317   307    -19ms[  -19ms] +/-   86ms
^- ns1.pulsation.fr              3  10   377   603  +6167us[+5818us] +/-  147ms
```

After adding the Amazon Server, the sync'd server at the top was the amazon server and it's estimated error was almost an order of magnitude smaller (not sure if that will matter at all)
```
bash-4.2$ chronyc sources -v
210 Number of sources = 5

  .-- Source mode  '^' = server, '=' = peer, '#' = local clock.
 / .- Source state '*' = current synced, '+' = combined , '-' = not combined,
| /   '?' = unreachable, 'x' = time may be in error, '~' = time too variable.
||                                                 .- xxxx [ yyyy ] +/- zzzz
||      Reachability register (octal) -.           |  xxxx = adjusted offset,
||      Log2(Polling interval) --.      |          |  yyyy = measured offset,
||                                \     |          |  zzzz = estimated error.
||                                 |    |           \
MS Name/IP address         Stratum Poll Reach LastRx Last sample               
===============================================================================
^* 169.254.169.123               3   4     7     2    -99ns[+1380us] +/-  511us
^- 50-205-244-108-static.hf>     2   6    17     1   +667us[ +667us] +/-   56ms
^- 159.203.82.102                4   6     7     3  -2911us[-1531us] +/-   44ms
^- las1.pawnode.com              3   6     7     2  -3055us[-1675us] +/-   47ms
^- pool-173-71-80-166.cmdnn>     2   6     7     2  +3149us[+4529us] +/-  140ms
```

I suppose if we want to avoid distributed time issues, we could always `make -t replace-rtl` after we copy the replace-rtl stuff back over.  I really dislike doing that if we don't have to.